test/tasks/task_flow_test.js

```diff
@@ -28,83 +37,146 @@ describe('task_flow_tests',()=>{
+      let bNodeController = bNode.mainController;
+      let bNodeCoreServer = bNode.coreServer;
+      let peerController = peer.mainController;
+      let peerCoreServer = peer.coreServer;
+
+      const stopTest = async ()=>{
+        await peerController.shutdownSystem();
+        peerCoreServer.disconnect();
+        await bNodeController.shutdownSystem();
+        bNodeCoreServer.disconnect();
+        resolve();
       };
+
+      const verifyResultPropagation = async (err) => {
+        assert.strictEqual(err, null);
+        const status = await peerController.getNode().taskManager().asyncGetTaskStatus(task.getTaskId());
+        assert.strictEqual(status, task.getStatus());
+        stopTest();
+      }
+
       // override the action response
-      peerController.getNode().overrideAction(constants.NODE_NOTIFICATIONS.RECEIVED_NEW_RESULT,{
-        execute : verifyPublish
-      });
+      const newAction = new VerifyAndStoreResultActionMock(peerController.getNode(), verifyResultPropagation);
+      peerController.getNode().overrideAction(constants.NODE_NOTIFICATIONS.RECEIVED_NEW_RESULT, newAction);
       // run the test
       // publish the task result
       bNodeController.getNode().execCmd(constants.NODE_NOTIFICATIONS.TASK_FINISHED, { task : task});
     });
   });
-});
 
-const generateDeployBundle = (num, isSuccess)=>{
-  let output = [];
-  let tasks = generateDeployTasks(num);
-  let status = constants.TASK_STATUS.SUCCESS;
-  if(!isSuccess){
-    status = constants.TASK_STATUS.FAILED;
-  }
-  tasks.forEach(t=>{
-    let resObj = {
-      taskId : t.getTaskId(),
-      status : status,
-      output : testUtils.getRandomByteArray(80),
-      delta : {index : 2, delta : testUtils.getRandomByteArray(20)},
-      usedGas : testUtils.getRandomInt(10000),
-      ethereumPayload : testUtils.getRandomByteArray(100),
-      ethereumAddress : testUtils.randLenStr(40),
-      signature : testUtils.getRandomByteArray(120),
-      blockNumber : testUtils.getRandomInt(100) ,
-      preCodeHash : testUtils.randLenStr(64),
-    };
-    let result = null;
-    if(isSuccess){
-      result = Result.DeployResult.buildDeployResult(resObj);
-    }else{
-      result = Result.FailedResult.buildFailedResult(resObj);
-    }
-    output.push({task : t, result : result});
+  it('#3 Should test w1 Publish a successful compute task and w2 receive it', async function(){
+    if(!tree['#3'] || !tree['all'] )
+      this.skip();
+    return new Promise(async resolve => {
+      // craete deploy task
+      let {task, result} = utils.generateComputeBundle(1,true)[0];
+      task.setResult(result);
+      // create all the boring stuff
+      let {bNode,peer} = await testBuilder.createTwo();
+      await testUtils.sleep(5000);
+      let bNodeController = bNode.mainController;
+      let bNodeCoreServer = bNode.coreServer;
+      let peerController = peer.mainController;
+      let peerCoreServer = peer.coreServer;
+
+      const stopTest = async ()=>{
```

Comment:
There isn't a way to make `stopTest` & `verifyResultPropagation` global functions that all the tests can call, instead of rewriting it every time?

Reply:
Good point, thanks! 
Did the appropriate change..